standard/expressions.md

```diff
@@ -36,6 +36,108 @@ Most of the constructs that involve an expression ultimately require the express
+- If the pattern is a case label ([§13.8.3](statements.md#1383-the-switch-statement)), then the scope of the variable is the associated *case block*.
+- Otherwise, the variable is part of the pattern that is the right-hand operand of the `is` operator ([§12.11.11](expressions.md#121111-the-is-operator)), and the variable’s scope is based on the construct immediately enclosing the `is` expression containing the pattern, as follows:
+  -  If the expression is in an expression-bodied lambda, the variable's scope is the body of the lambda.
+  -  If the expression is in an expression-bodied method or property, the variable's scope is the body of the method or property.
+  -  If the expression is in a `when` clause of a `catch` clause, the variable's scope is that `catch` clause.
+  -  If the expression is in an *iteration_statement*, the variable's scope is just that statement.
+  -  If the expression is in a *constructor_initializer*, the variable's scope is the part of that *constructor_initializer* following the expression, and the body of the constructor.
+  -  If the expression is in a *variable_initializer* of a field, the variable's scope is the part of that *variable_initializer* following the expression.
+  -  If the expression is in a *query_expression* that is translated into the body of a lambda, the variable's scope is that *query_expression*.
+  -  Otherwise if the expression is in some other statement form, the variable's scope is the scope containing the statement.
+
+For the purpose of determining the scope, an *embedded_statement* is considered to be in its own scope.
+
+#### §declaration-pattern-new-clause Declaration pattern
+
+A *declaration_pattern* is used to test that an expression has a given type and if the test succeeds, to cast that expression’s value to that type. 
+
+```antlr
+declaration_pattern
+    : type simple_designation
+    ;
+simple_designation
+    : single_variable_designation
+    ;
+single_variable_designation
+    : identifier
+    ;
+```
+
+The runtime type of the pattern context expression is tested against the *type* in the pattern. If it is of that runtime type (or some subtype), the result of the `is` operator is `true`. A pattern context expression with value `null` never tests true for this pattern.
+
+Given a pattern context expression (§patterns-new-clause) *e*, if the *simple_designation* is the identifier `_` (see §discards-new-clause) the value of *e* is not bound to anything. (Although a declared variable with the name `_` may be in scope at that point, that named variable is not seen in this context.) If *designation* is any other identifier, a local variable ([§10.2.8](variables.md#1028-local-variables)) of the given type named by the given identifier is introduced, and that local variable is definitely assigned ([§10.4](variables.md#104-definite-assignment)) with the value of the pattern context expression when the result of the pattern-matching operation is true.
```

Comment:
This form of description will not generalize to nested patterns gracefully.

Reply:
I'll take another look.

Reply:
@gafter I'm actually writing up the v8 pattern extensions diffs spec this week, and looking at them, I *don't* see that they complicate things in this regard. In my v7 text above, I have 

```ANTLR
simple_designation
    : single_variable_designation
    ;
```

which in v8 will become

```ANTLR
simple_designation
    : single_variable_designation
    | discard_designation
    ;
```
with the rule *single_variable_designation* remaining unchanged as *identifier*. I don't see that nested patterns impact this part of the grammar.